targettests/execution/gen_device_code.cpp

```diff
@@ -20,27 +20,41 @@ struct baz {
   void operator()(cudaq::qubit &q) __qpu__ { z(q); }
 };
 
-struct bar {
+struct bah {
+  void operator()(cudaq::qubit &q) __qpu__ { h(q); }
+};
+
+struct bar1 {
   void operator()(cudaq::qubit &q) __qpu__ {
     bax{}(q);
     baz{}(q);
   }
 };
 
+struct bar2 {
+  void operator()(cudaq::qubit &q) __qpu__ {
+    bah{}(q);
+    bax{}(q);
+    bar1{}(q);
+  }
+};
+
 struct foo {
   void operator()() __qpu__ {
     cudaq::qubit q;
-    bar{}(q);
+    bar2{}(q);
+    bax{}(q);
     mz(q);
   }
 };
 
 int main() {
-  cudaq::sample(foo{});
   std::cout << cudaq::get_quake(foo{}) << "\n";
   return 0;
 }
 // CHECK-LABEL:   func.func @__nvqpp__mlirgen__foo
-// CHECK-LABEL:   func.func @__nvqpp__mlirgen__bar
+// CHECK-LABEL:   func.func @__nvqpp__mlirgen__bar2
+// CHECK-LABEL:   func.func @__nvqpp__mlirgen__bah
 // CHECK-LABEL:   func.func @__nvqpp__mlirgen__bax
+// CHECK-LABEL:   func.func @__nvqpp__mlirgen__bar1
```

Comment:
Are these guaranteed to be ordered in any way by the call graph iterator? Might want to use `CHECK-DAG` to make the test imperious to random iterator traversal changes.

Reply:
This should be guaranteed by the 'depth first' traversal iterator that we chose. I think it might be worth checking the deterministic behavior that we expected in this case.